### PR TITLE
Add custom schema fields to Amazon attach cloud volume form

### PR DESCRIPTION
**Pull request by MelsHyrule**
Pre-requisite / companion PR to [1141](https://github.com/ManageIQ/manageiq-api/pull/1141) and [8205](https://github.com/ManageIQ/manageiq-ui-classic/pull/8205). 

Part of the process for upgrading [_attach.html.haml](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/views/vm_common/_attach.html.haml#L33) and [_detach.html.haml](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/views/vm_common/_detach.html.haml#L25) to react as per https://github.com/ManageIQ/manageiq-ui-classic/issues/7603.

### CODE REVIEW

**Comment by MelsHyrule** [comment]
@miq-bot add-reviewer @kavyanekkalapu
@miq-bot add-reviewer @kbrock

File: app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb [hunk 1]
```diff
@@ -298,6 +298,21 @@ def params_for_update
...
+  def params_for_attach
+    {
+      :fields => [
+        {
+          :component    => 'text-field',
+          :name         => 'device_mountpoint',
+          :id           => 'device_mountpoint',
+          :label        => _('Device Mountpoint (Requiered)'),
+          :isRequired   => true,
+          :validate     => [{:type => 'required'}],
```
Comment (agrare) [inline app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb:310]:
```suggestion
          :component  => 'text-field',
          :name       => 'device_mountpoint',
          :id         => 'device_mountpoint',
          :label      => _('Device Mountpoint (Requiered)'),
          :isRequired => true,
          :validate   => [{:type => 'required'}],
```

**Comment by agrare** [comment]
Just the rubocop issues, I think we also need to define this method on the rest of the providers which support cloud_volume_attach

**Review by kbrock — COMMENTED** [review]
yea. this looks good.

As far as I can tell, lets fixup the hash alignment and then merge

File: app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb [hunk 2]
```diff
@@ -298,6 +298,21 @@ def params_for_update
...
   end
 
+  def params_for_attach
+    {
+      :fields => [
+        {
+          :component  => 'text-field',
+          :name       => 'device_mountpoint',
+          :id         => 'device_mountpoint',
+          :label      => _('Device Mountpoint (Requiered)'),
```
Comment (Fryguy) [inline app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb:308]: typo

```suggestion
          :label      => _('Device Mountpoint (Required)'),
```

Comment (Fryguy) [reply app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb:308]: Aside, do we typically add `"Required"` to required fields, or do we let the form required speak for itself (either with red highlighting or a `*` or something)?

Comment (Fryguy) [reply app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb:308]: Looking at the other parsms_for methods we don't indicate it so, I'm thinking drop the wording...

```suggestion
          :label      => _('Device Mountpoint'),
```

**Pull request merged**